docs/spec/spec.md

```diff
@@ -0,0 +1,2314 @@
+        returns targetType|ClientError;
+```
+
+```ballerina
+http:Client httpClient = check new ("https://www.example.com");
+string response = check httpClient->post("/some/endpoint",
+   {
+       name: "foo",
+       age: 25,
+       address: "area 51"
+   },
+   headers = {
+       "my-header": "my-header-value"
+   }
+   mediaType = "application/json",
+);
+```
+
+###### 2.4.2.2 Non Entity body methods
+
+GET, HEAD, OPTIONS methods are considered as non entity body methods. These remote functions does not contains 
+RequestMessage, but the header map an optional param.
+
+
+```ballerina
+# The head() function can be used to send HTTP HEAD requests to HTTP endpoints.
+remote isolated function head(string path, map<string|string[]>? headers = ()) returns Response|ClientError;
+
+# The get() function can be used to send HTTP GET requests to HTTP endpoints.
+remote isolated function get( string path, map<string|string[]>? headers = (), TargetType targetType = <>)
+        returns  targetType|ClientError;
+
+# The options() function can be used to send HTTP OPTIONS requests to HTTP endpoints.
+remote isolated function options( string path, map<string|string[]>? headers = (), TargetType targetType = <>)
+        returns  targetType|ClientError;
+```
+
+````ballerina
+http:Client httpClient = check new ("https://www.example.com");
+map<string|string[]> headers = {
+   "my-header": "my-header-value",
+   "header-2": ["foo", "bar"]
+};
+string resp = check httpClient->get("/data", headers);
+````
+
+###### 2.4.2.3 Forward/Execute methods
+
+In addition to the standard HTTP methods, `forward` function can be used to proxy an inbound request using the incoming 
+HTTP request method. Also `execute` remote function is useful to send request with custom HTTP verbs such as `move`, 
+`copy`, ..etc.
+
+
+```ballerina
+# Invokes an HTTP call with the specified HTTP verb.
+remote isolated function execute(string httpVerb,  string path, RequestMessage message, 
+        map<string|string[]>? headers = (), string? mediaType = (), TargetType targetType = <>)
+        returns targetType|ClientError;
+
+# The forward() function can be used to invoke an HTTP call with inbound request's HTTP verb
+remote isolated function forward(string path, Request request, TargetType targetType = <>)
+        returns  targetType|ClientError;
+```
+
+###### 2.4.2.4 HTTP2 additional methods
+Following are the HTTP2 client related additional remote functions to deal with promises and responses.
+
+```ballerina
+
+# Submits an HTTP request to a service with the specified HTTP verb.
+# The submit() function does not give out a http:Response as the result.
+# Rather it returns an http:HttpFuture which can be used to do further interactions with the endpoint.
+remote isolated function submit(string httpVerb, string path, RequestMessage message)
+    returns HttpFuture|ClientError;
+
+# This just pass the request to actual network call.
```

Comment:
Instead of `This just pass the request to actual network call.` maybe we can say `Passes the request to actual network call.`